server/views/cms/rest.py

```diff
@@ -0,0 +1,67 @@
+import logging
+from flask import jsonify, request
+import requests
+import html
+
+from server import app
+from server.util.request import api_error_handler
+
+logger = logging.getLogger(__name__)
+BASE_URL = app.config['PAYLOAD_API_URL']
+API_KEY = app.config['PAYLOAD_API_KEY']
+
+
+@app.route('/api/cms/fetch-page-content', methods=['GET'])
+@api_error_handler
+def api_fetch_page_content():
+    application_name = html.escape(request.headers.get('cs-app') or '')
+    url = f"{BASE_URL}/{application_name}-pages"
```

Comment:
1. Feels like application_name is required i.e. if empty we should throw an error.
2. Don't we need to format `BASE_URL` to avoid double `//` in case `BASE_URL` already has `/` at the end?